### PR TITLE
build(sample): Bump sample app to React Native 0.64

### PR DESCRIPTION
**Pull request by jennmueng**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring

## :scroll: Description

Bumps the sample app to the latest React Native version 0.64.

## :bulb: Motivation and Context

Users have been having issues with this latest version.

## :green_heart: How did you test it?
Ran both iOS and Android emulators.

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] All tests passing (e2e tests passed locally)
- [x] No breaking changes

#skip-changelog

### CODE REVIEW

File: sample/ios/sample.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -868,6 +868,7 @@
 				COPY_PHASE_STRIP = NO;
 				ENABLE_STRICT_OBJC_MSGSEND = YES;
 				ENABLE_TESTABILITY = YES;
+				"EXCLUDED_ARCHS[sdk=iphonesimulator*]" = "arm64 ";
```
Comment (philipphofmann) [inline sample/ios/sample.xcodeproj/project.pbxproj:871]: `l`: Why do we need this?

File: sample/ios/Podfile [hunk 2]
```diff
@@ -5,14 +5,22 @@ platform :ios, '10.0'
 
 target 'sample' do
   config = use_native_modules!
-  use_react_native!(:path => config["reactNativePath"])
+  use_react_native!(
+    :path => config["reactNativePath"],
+    # to enable hermes on iOS, change `false` to `true` and then install pods
+  	:hermes_enabled => false
```
Comment (philipphofmann) [inline sample/ios/Podfile:11]: `l`: Why do we need to disable it? Can we maybe add this info as a comment?

**Review by philipphofmann — APPROVED** [review]
LGTM.

File: sample/ios/sample.xcodeproj/project.pbxproj (hunk 1 above)
Comment (marandaneto) [reply sample/ios/sample.xcodeproj/project.pbxproj:871]: I guess we have something similar on Flutter https://github.com/getsentry/sentry-dart/issues/348

**Comment by jennmueng** [comment]
Investigating the failing e2e test...

File: sample/ios/Podfile (hunk 2 above)
Comment (jennmueng) [reply sample/ios/Podfile:11]: This is the disabled-by-default from React Native, we should actually have it as true for the sample app.

**Pull request merged**